test/internal/laws/forall.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var jsc = require('jsverify');
+var R = require('ramda');
+
+//  forall :: StrMap Function -> StrMap Function
+module.exports = function forall(laws) {
+  return Object.keys(laws).reduce(function(result, name) {
+    result[name] = function() {
+      var args = Array.prototype.slice.call(arguments);
+      function pred() {
+        var pair = laws[name].apply(null, arguments);
+        var lhs = pair[0];
+        var rhs = pair[1];
+        return R.equals(lhs, rhs);
```

Comment:
That would be nice!
I don't understand why are you returning `lhs` and `rhs` as `Tuple` when `equals` function could be passed as argument (which will take as arguments the lhs and rhs)